jax/config.py

```diff
@@ -90,8 +94,12 @@ def _read(self, name):
   def add_option(self, name, default, opt_type, meta_args, meta_kwargs):
     if name in self.values:
       raise Exception("Config option {} already defined".format(name))
+    update_hook = meta_kwargs.pop("update_hook", None)
```

Comment:
Maybe best not to mutate the passed-in `meta_kwargs` argument, unless I'm missing some context for why it's necessary?

Reply:
As far as I can see the purpose of `meta_kwargs` is to pass kwargs to absl, but we don't want to pass this argument to absl.

Reply:
Maybe shallow copy before modifying?

Reply:
Looks like this always comes via `**kwargs` in `config.DEFINE_*`, so I think mutation is fine.

Reply:
A function mutating passed-in args is an error-prone pattern even if it's not _currently_ a bug given the callers in this file. (If this were clearly an internal-only method, like if it were named `_add_options`, then it'd be less error-prone, but still not great.)

How about just adding a `meta_kwargs = meta_kwargs.copy()` or similar?

Reply:
Done.

Reply:
Actually I had to move the parsing of this argument into the callers to make pytype happy (fine, whatever...) which makes this moot, since the `kwargs` are now ephemeral.